Tests/powershell-yaml.Tests.ps1

```diff
@@ -711,11 +712,19 @@ int64: 9223372036854775807
             $result.bigInt | Should -BeOfType System.Numerics.BigInteger
         }
 
+        It "Should round-trip decimals with trailing 0" {
+            $result = ConvertFrom-Yaml -Yaml $value
+            $result.decimal | Should -Be ([decimal]3.10)
```

Comment:
I'd add a test for `.Scale` here.

The `[decimal]3.10` is not really working in PowerShell as you might expect - this firstly creates a `double` literal, and then that is converted to `decimal` object. So in reality, the `[decimal]3.10` would have Scale=1 and will print (via ToString e.g.) as `3.1`. If you construct `decimal` via explicit Parsing though, it will correctly have Scale=2. See:

```pwsh
$scaletest = '3.9999999999999990'
[decimal]3.9999999999999990
# prints 4
[decimal]$scaletest
# prints 4
[decimal]::Parse($scaletest, [cultureinfo]::InvariantCulture)
# prints `3.9999999999999990`
```